clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -221,7 +221,7 @@ public void send(ClientRequest request, long now) {
                     builder.latestAllowedVersion());
             AbstractRequest abstractRequest = request.requestBuilder().build(version);
             if (!futureResp.requestMatcher.matches(abstractRequest))
-                throw new IllegalStateException("Request matcher did not match next-in-line request " + abstractRequest);
+                throw new IllegalStateException("Request matcher did not match next-in-line request " + abstractRequest + " with prepared response " + futureResp.responseBody);
```

Comment:
nit: break up long line

Reply:
@ableegoldman Sorry I missed these comments. Will address in the pt.3 PR.